File: prop-all/src/test/java/io/pleo/prop/objects/SamePropertyAsComplexObjects.java
package io.pleo.prop.objects;

import javax.inject.Inject;
import javax.inject.Named;

import io.pleo.prop.core.Prop;

public class SamePropertyAsComplexObjects {
  private Prop<InjectedObject> myComplexObjectProp;

  @Inject
  public SamePropertyAsComplexObjects(@Named("io.pleo.test.prop1") Prop<InjectedObject> myComplexObjectProp) {
    this.myComplexObjectProp = myComplexObjectProp;
  }

  public Prop<InjectedObject> getMyComplexObjectProp() {
    return myComplexObjectProp;
  }
}

File: prop-all/src/test/kotlin/io/pleo/prop/objects/SamePropertyAsComplexObjects.kt
package io.pleo.prop.objects

import com.google.inject.Inject
import com.google.inject.name.Named
import io.pleo.prop.core.Prop

class SamePropertyAsComplexObjects
@Inject
constructor(
    @Named("io.pleo.test.prop1") val myComplexObjectProp: Prop<InjectedObject>,
)
